<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve the distribution of genes for given organism in genbank — gene.sampling.retrieve • phruta</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve the distribution of genes for given organism in genbank — gene.sampling.retrieve"><meta property="og:description" content="This function retrieves the gene names for given organism in genbank. This function is
useful to explore the genes that are widely sampled for a given taxon
in genbank. However, note that the performance of gene.sampling.retrieve
is entirely dependant on the quality of submissions to genbank. For instance, if
most of the sequences in genbank (nucleotide) don't have information on the
sequenced region, this function won't provide reliable estimates of the sampling
frequency of each gene in the database. We recommend using this function with caution,
and only when there is no additional information on the genes that are widely sampled
for the target group(s) of interest."><meta property="og:image" content="https://cromanpa94.github.io/phruta/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phruta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/what_is_phruta.html">What is `phruta`?</a>
    </li>
    <li>
      <a href="../articles/Using_phruta.html">Using the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Exporting_data_phruta.html">To export or not export `phruta` outputs</a>
    </li>
    <li>
      <a href="../articles/Phylogenetics_phruta.html">Phylogenetics with the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Phruta_advanced.html">Advanced methods in the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Future_phruta.html">Future directions of the `phruta` R package</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/cromanpa94/phruta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve the distribution of genes for given organism in genbank</h1>
    <small class="dont-index">Source: <a href="https://github.com/cromanpa94/phruta/blob/HEAD/R/gene.sampling.retrieve.R" class="external-link"><code>R/gene.sampling.retrieve.R</code></a></small>
    <div class="hidden name"><code>gene.sampling.retrieve.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function retrieves the gene names for given organism in genbank. This function is
useful to explore the genes that are widely sampled for a given taxon
in genbank. However, note that the performance of <code>gene.sampling.retrieve</code>
is entirely dependant on the quality of submissions to genbank. For instance, if
most of the sequences in genbank (nucleotide) don't have information on the
sequenced region, this function won't provide reliable estimates of the sampling
frequency of each gene in the database. We recommend using this function with caution,
and only when there is no additional information on the genes that are widely sampled
for the target group(s) of interest.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gene.sampling.retrieve</span><span class="op">(</span><span class="va">organism</span>, speciesSampling <span class="op">=</span> <span class="cn">TRUE</span>, npar <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>organism</dt>
<dd><p>A vector of taxonomic groups (character).</p></dd>


<dt>speciesSampling</dt>
<dd><p>Whether the results should be at the species-level (logical).</p></dd>


<dt>npar</dt>
<dd><p>Number of simultaneous searches (character; optional). The default 2 is
generally fast enough and does not exceed the maximum number of connections
to genbank.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>data.frame</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">test.spp</span> <span class="op">&lt;-</span> <span class="fu">gene.sampling.retrieve</span><span class="op">(</span>organism <span class="op">=</span> <span class="st">"Puma"</span>, speciesSampling <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">test.pop</span> <span class="op">&lt;-</span> <span class="fu">gene.sampling.retrieve</span><span class="op">(</span>organism <span class="op">=</span> <span class="st">"Puma"</span>, speciesSampling <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer></div>

  


  

  </body></html>

