<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Phylogenetic Reconstruction and Time-dating • phruta</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Phylogenetic Reconstruction and Time-dating">
<meta property="og:description" content="The phruta R package is designed to simplify the basic 
            phylogenetic pipeline. Specifically, all code is run within 
            the same program and data from intermediate steps are saved in 
            independent folders. Furthrmore, all code is run within the 
            same environment which increases the reproducibility of your analysis. 
            phruta retrieves gene sequences, combines newly downloaded and local 
            gene sequences, and performs sequence alignments. ">
<meta property="og:image" content="https://cromanpa94.github.io/phruta/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">phruta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Brief_Intro_phruta.html">Introduction to the `phruta` R package</a>
    </li>
    <li>
      <a href="articles/Phylogenetics_phruta.html">Phylogenetics with the `phruta` R package</a>
    </li>
    <li>
      <a href="articles/Phruta_advanced.html">Advanced methods in the `phruta` R package</a>
    </li>
    <li>
      <a href="articles/Future_phruta.html">Future directions of the `phruta` R package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cromanpa94/phruta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<p><!-- badges: start --> <a href="https://codecov.io/gh/cromanpa94/phruta?branch=main" class="external-link"><img src="https://codecov.io/gh/cromanpa94/phruta/branch/main/graph/badge.svg" alt="Codecov test coverage"></a> <a href="https://github.com/cromanpa94/phruta/actions" class="external-link"><img src="https://github.com/cromanpa94/phruta/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a> <a href="https://lifecycle.r-lib.org/articles/stages.html#maturing" class="external-link"><img src="https://img.shields.io/badge/lifecycle-maturing-blue.svg"></a><br><a href="https://github.com/cromanpa94/phruta" class="external-link"><img src="https://img.shields.io/github/languages/code-size/cromanpa94/phruta.svg"></a> <a href="https://www.codefactor.io/repository/github/cromanpa94/phruta" class="external-link"><img src="https://www.codefactor.io/repository/github/cromanpa94/phruta/badge" alt="CodeFactor"></a> <!-- badges: end --></p>
<div class="section level1">
<div class="page-header"><h1 id="phruta-">
<code>phruta</code> <a href="https://cromanpa94.github.io/phruta"><img src="reference/figures/logo.png" align="right" height="300"></a>
<a class="anchor" aria-label="anchor" href="#phruta-"></a>
</h1></div>
<div class="section level3">
<h3 id="assembling-phylogenetic-trees-from-taxonomic-names">Assembling phylogenetic trees from taxonomic names.<a class="anchor" aria-label="anchor" href="#assembling-phylogenetic-trees-from-taxonomic-names"></a>
</h3>
</div>
<div class="section level2">
<h2 id="what-is-phruta">What is <code>phruta</code>?<a class="anchor" aria-label="anchor" href="#what-is-phruta"></a>
</h2>
<p>The <code>phruta</code> R package is designed to simplify the basic phylogenetic pipeline. Specifically, all code is run within the same program and data from intermediate steps are saved in independent folders. Furthrmore, all code is run within the same environment which increases the reproducibility of your analysis. <code>phruta</code> retrieves gene sequences, combines newly downloaded and local gene sequences, and performs sequence alignments.</p>
<p><code>phruta</code> is also able to perform basic phylogenetic inference under <code>RAxML</code> on the resulting sequence alignments. The current release allows users to conduct tree dating based on secondary calibrations. <code>phruta</code> is essentially a wrapper for alternative R packages and software.</p>
</div>
<div class="section level2">
<h2 id="who-should-consider-using-phruta">Who should consider using <code>phruta</code>
<a class="anchor" aria-label="anchor" href="#who-should-consider-using-phruta"></a>
</h2>
<p>The main functions in the <code>phruta</code> R package allow for a quick mining and curation of GenBank sequences. This package is designed for students and researchers interested in generating species-level genetic datasets for particular sets of taxa. Specifically, if you have a clade or group of species in mind, <code>phruta</code> will help you to assemble a molecular dataset with information available in GenBank.</p>
</div>
<div class="section level2">
<h2 id="why-use-phruta">Why use <code>phruta</code>?<a class="anchor" aria-label="anchor" href="#why-use-phruta"></a>
</h2>
<p><code>phruta</code> simplifies the phylogenetic pipeline, increases reproducibility, and helps organizing information used to infer molecular phylogenies.</p>
</div>
<div class="section level2">
<h2 id="how-is-phruta-different-from-other-software">How is <code>phruta</code> different from other software?<a class="anchor" aria-label="anchor" href="#how-is-phruta-different-from-other-software"></a>
</h2>
<p><code>phruta</code> has two core functions. The main applications of these functions is briefly outlined below:</p>
<ul>
<li><p><code>sq.retrieve*</code>: This function downloads sequences from genbank (nucleotide database) for particular taxa (taxonomic groups or particular species) and a list of genes.</p></li>
<li><p><code><a href="reference/sq.curate.html">sq.curate()</a></code>: After sequences are downloaded from genbank, this function curates sequences within each of the examined genes by detecting sequence outliers and by using taxonomic information.</p></li>
</ul>
<p>In addition to these two main functions, users will be able to align the downloaded sequences, infer phylogenetic trees, and calibrate phylogenies using additional functions in <code>phruta</code>.</p>
</div>
<div class="section level2">
<h2 id="installing-phruta">Installing <code>phruta</code>
<a class="anchor" aria-label="anchor" href="#installing-phruta"></a>
</h2>
<p><code>phruta</code> is currently only available through <code>GitHub</code>. It can be easily installed using the following code.</p>
<pre><code><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span> 
<span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"cromanpa94/phruta"</span><span class="op">)</span></code></pre>
<p>Please make sure that the R packages <code>msa</code>, <code>DECIPHER</code>, <code>Biostrings</code>, and <code>odseq</code> are correctly installed. Finally, if you’re interested in using the development version of <code>phruta</code>, feel free to use the following code:</p>
<pre><code><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span> 
<span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"cromanpa94/phruta"</span>, subdir <span class="op">=</span> <span class="st">"devB"</span><span class="op">)</span></code></pre>
</div>
<div class="section level2">
<h2 id="installing-raxml-">Installing RAxML <a name="paragraph1"></a>
<a class="anchor" aria-label="anchor" href="#installing-raxml-"></a>
</h2>
<p><code>RAxML</code> can be easily installed to the <code>PATH</code> using one of the two lines below in <code>conda</code>:</p>
<p><code>{bash eval=FALSE} conda install -c bioconda/label/cf201901 raxml</code></p>
<p><code>{bash eval=FALSE} conda install -c bioconda raxml</code></p>
<p>Open <code>R</code> and make sure that the following line doesn’t throw an error.</p>
<p><code>{r eval=FALSE} system("raxmlHPC")</code></p>
<p>You may also want to check if <code>RAxML</code> is called using <code>raxmlHPC</code> or something else. This can be set when running <code>tree.raxml</code> using the <code>raxml_exec</code> argument.</p>
<p>Finally, note that <code>RStudio</code> sometimes has issues finding <em>stuff</em> in the path while using <code><a href="https://rdrr.io/r/base/system.html" class="external-link">system()</a></code>.</p>
<p>With issues, if you’re running macOS, try running RStudio from the command line. Close RStudio, open the terminal, and type:</p>
<p><code>{bash eval=FALSE} open /Applications/RStudio.app</code></p>
</div>
<div class="section level2">
<h2 id="installing-pathd-8-and-treepl-">Installing PATHd-8 and treePL <a name="paragraph2"></a>
<a class="anchor" aria-label="anchor" href="#installing-pathd-8-and-treepl-"></a>
</h2>
<p>There are excellent guides for installing <code>PATHd-8</code> and <code>treePL</code>. Here, I summarize two potentially relevant options.</p>
<p>First, you can use <a href="https://github.com/bomeara/phydocker/blob/master/Dockerfile" class="external-link">Brian O’Meara’s</a> approach for installing <code>PATHd-8</code>. I summarized the code <a href="https://gist.github.com/cromanpa94/a43bc710a17220f71d796d6590ea7fe4" class="external-link">here</a>.</p>
<p>Second, you can use homebrew to install <code>treePL</code>, thanks to Jonathan Chang.</p>
<p><code>{bash eval = F} brew install brewsci/bio/treepl</code></p>
</div>
<div class="section level2">
<h2 id="running-phruta-from-rstudio-while-using-macos">Running <code>phruta</code> from <code>Rstudio</code> while using <code>MacOS</code>?<a class="anchor" aria-label="anchor" href="#running-phruta-from-rstudio-while-using-macos"></a>
</h2>
<p>Only if you’re interested in running phylogenetic analyses, please make sure you open <code>RStudio</code> using the following code from the terminal:</p>
<p><code>{bash eval=FALSE} open /Applications/RStudio.app</code></p>
</div>
<div class="section level2">
<h2 id="dedication">Dedication<a class="anchor" aria-label="anchor" href="#dedication"></a>
</h2>
<p>My package is dedicated to my mom and every single Black woman. I still have lots of things to learn from you. You will always have all my admiration.</p>
<p>The <a href="https://www.flickr.com/photos/gufomusike/3462117620/in/photolist-6NFiPi-xoLbca-FtC6yJ-4nk6wS-x2AZV-b3MUv8-e2B7qj-4uCwwa-e3PJxi-2ePGmUM-b2wBVi-obHf1x-5iP26P-4juoE6-z881E-z88t3-9GmTbQ-dGvrFe-22APdBs-p2t5Zv-8DWQw8-6fAJ2G-7jQhu2-7LEkkL-7vBdyF-jTdXSR-kcntD1-aWGfnx-bk59CK-5JfhKt-6gWfX7-reVehy-bjk7Ki-2xnGjv-dLJbq9-e3VjY3-ugz6U-FGVagm-iqVRuD-YE5pLe-2kPkt84-2kHhswd" class="external-link">logo</a> features a Palenquera in Cartagena (Colombia). For many folks, Palenqueras are just the Black woman ones who sell fruits in particular Colombian turistic areas. However, palenqueras and Palenque are central to Black identity in Colombia, Latin America, and America. <a href="https://en.wikipedia.org/wiki/San_Basilio_de_Palenque" class="external-link">“Palenque was the first free African town in the Americas”</a>.</p>
<p><em>Fruta</em> is the Spanish word for <em>Fruit</em>. English <em>ph</em> sounds the same as <em>F</em> in Spanish. In <code>phruta</code>, <em>ph</em> is relative to phylogenetics. I pronounce <code>phruta</code> just as <em>fruta</em> in Spanish.</p>
</div>
<div class="section level2">
<h2 id="additional-resources">Additional resources<a class="anchor" aria-label="anchor" href="#additional-resources"></a>
</h2>
<p>More details about the functions implemented in <code>phruta</code> can be found in the different vignettes associated with the package or in our <a href="https://cromanpa94.github.io/phruta/">website</a>.</p>
</div>
<div class="section level2">
<h2 id="contributing">Contributing<a class="anchor" aria-label="anchor" href="#contributing"></a>
</h2>
<p>Please see our <a href="CONTRIBUTING">contributing guide</a>.</p>
</div>
<div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<p>Please see the package <a href="DESCRIPTION">DESCRIPTION</a> for package authors.</p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/cromanpa94/phruta/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/cromanpa94/phruta/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing phruta</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-1696-4886" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
