<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>phruta • phruta</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/Yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="phruta">
<meta property="og:description" content="phruta">
<meta property="og:image" content="https://cromanpa94.github.io/phruta/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phruta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/phruta.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://twitter.com/cromanpa" class="external-link">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/cromanpa94/phruta" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>phruta</h1>
                        <h4 data-toc-skip class="author">Cristian
Román-Palacios</h4>
            
            <h4 data-toc-skip class="date">First created on 2021-07-30.
Updated on 2022-06-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/cromanpa94/phrutavignettes/phruta.Rmd" class="external-link"><code>vignettes/phruta.Rmd</code></a></small>
      <div class="hidden name"><code>phruta.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<ol style="list-style-type: decimal">
<li><a href="#intro">What is <code>phruta</code></a></li>
<li><a href="#paragraph0">Functions in <code>phruta</code></a></li>
</ol>
<div class="section level3">
<h3 id="what-is-phruta">What is <code>phruta</code> <a name="intro"></a><a class="anchor" aria-label="anchor" href="#what-is-phruta"></a>
</h3>
<p>The <code>phruta</code> package is designed to simplify the basic
phylogenetic pipeline in <code>R</code>. <code>phruta</code> is designed
to allow scientists from different backgrounds to assemble their own
reproducible phylogenies with as minimal code as possible. All code in
<code>phruta</code> is run within the same program (<code>R</code>) and
data from intermediate steps are either stored to the environment or
exported locally in independent folders. All code in <code>phruta</code>
is run within the same environment, an aspect that increases the
reproducibility of your analysis. <code>phruta</code> looks for
potentially (phylogenetically) relevant gene regions for a given set of
taxa, retrieves gene sequences, could combine newly downloaded and local
gene sequences, performs sequence alignment, phylogenetic inference, and
tree dating. <code>phruta</code> is largely a wrapper for alternative
<code>R</code> packages and software.</p>
</div>
<div class="section level3">
<h3 id="functions-in-phruta">Functions in <code>phruta</code> <a name="paragraph0"></a><a class="anchor" aria-label="anchor" href="#functions-in-phruta"></a>
</h3>
<p>The current release of <code>phruta</code> includes a set of eight
major functions. All eight functions form a pipeline within
<code>phruta</code> to output a time-calibrated phylogeny. However,
users interested in using their own files at any stage can run each
function independently.</p>
<p>Note that all the functions for which their primary output are
sequences (aligned or unaligned) are listed under <code>sq.*</code>. All
the files that output phylogenies (time-calibrated or not) are listed
under <code>tree.*</code>.</p>
<ul>
<li><p>First, the distribution of gene sampled for a given organism or
set of taxa can be explored using the <code>acc.gene.sampling</code>
function. This function will return a table that summarizes either the
distribution of genes sampled for the search term in general or
specifically across species.</p></li>
<li><p>Second, given a list of target organisms, users can retrieve a
list of accession numbers that are relevant to their search using
<code>acc.table.retrieve</code>. Instead of directly downloading
sequences from genbank (see <code>sq.retrieve.direct</code> below),
retrieving accession numbers allow users to have more control over the
sequences that are being used in the analyses. Note that users can also
curate the content of the dataset obtained using
<code>sq.retrieve.direct</code>.</p></li>
<li><p>Third, users should download gene sequences. Sequences can be
download using the <code>sq.retrieve.indirect</code> from the accession
numbers retrieved before using the <code>acc.table.retrieve</code>
function. This is the preferred option within <code>phruta</code>.
Additionally, users can directly download gene sequences using the
<code>sq.retrieve.direct</code> function. Both
<code>sq.retrieve.indirect</code> and <code>sq.retrieve.direct</code>
functions save gene sequences in <code>fasta</code> files that will be
located in a new directory named <code>0.Sequences</code>.</p></li>
<li><p>Fourth, <code>sq.add</code> allows users to include local
sequences to those retrieved from genbank in the previous step. This
function saves all the resulting <code>fasta</code> files in two
directories, combined sequences in <code>0.Sequences</code> and local
sequences in <code>0.AdditionalSequences</code> (originally downloaded
sequences are moved to <code>0.0.OriginalDownloaded</code> at this
step). Note that <code>sq.add</code> is optional.</p></li>
<li><p>Fifth, the <code>sq.curate</code> function filters out unreliable
sequences based on information listed in genbank (e.g. PREDICTED) and on
taxonomic information provided by the user. Specifically, this function
retrieves taxonomic information from the Global Biodiversity Information
Facility (GBIF) database’s taxonomic backbone (see alternatives in the
advanced vignette to <code>phruta</code>). If a given species belongs to
a non-target group, this species is dropped from the analyses. This
function automatically corrects taxonomy and renames sequences.</p></li>
<li><p>Sixth, <code>sq.aln</code> performs multiple sequence alignment
on <code>fasta</code> files. Currently, <code>phruta</code> uses the <a href="http://www2.decipher.codes/" class="external-link"><code>DECIPHER</code> R package</a>,
here. This package allows for adjusting sequence orientation and masking
(removing ambiguous sites).</p></li>
</ul>
<p>The final two functions in <code>phruta</code> focus on tree
inference and dating. These two functions depend on external software
that needs to be installed (<strong>and tested</strong>) before running.
Please make sure both <code>RAxML</code> and <code>PATHd-8</code> or
<code>treePL</code> are installed and can be called within
<code>R</code> using the <code><a href="https://rdrr.io/r/base/system.html" class="external-link">system()</a></code> function. Note that you
can choose between <code>PATHd-8</code> and <code>treePL</code>. More
details on how to install <code>RAxML</code> are provided in the
phylogenetic vignette of <code>phruta</code>. Similarly, we provide
details on how to install <code>PATHd-8</code> and <code>treePL</code>
in the same vignette.</p>
<ul>
<li><p>Seventh, the <code>tree.raxml</code> function allows users to
perform tree inference under <code>RAxML</code> for sequences in a given
folder. This is a wrapper to <code><a href="https://rdrr.io/pkg/ips/man/raxml.html" class="external-link">ips::raxml</a></code> and each of the
arguments can be customized. The current release of <code>phruta</code>
can manage both partitioned and unpartitioned analyses. Starting and
constrained trees are allowed.</p></li>
<li><p>Eight, <code>tree.dating</code> enables users to perform
time-calibrations of a given phylogeny using
<code><a href="https://rdrr.io/pkg/geiger/man/congruify.phylo.html" class="external-link">geiger::congruify.phylo</a></code>. <code>phruta</code> includes a
basic set of comprehensively sampled, time-calibrated phylogenies that
are used to extract secondary calibrations for the target phylogeny.
Note that sampling in those phylogenies can be examined using
<code>data(SW.phruta)</code>. Please make sure you have at least
<strong>two</strong> groups in common with each of the phylogenies.
Similarly, users can choose to run either <code>PATHd-8</code> or
<code>treePL</code>.</p></li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  <script data-goatcounter="https://phruta.goatcounter.com/count" async src="//gc.zgo.at/count.js"></script>
</body>
</html>
