<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Advanced methods in the `phruta` R package • phruta</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Advanced methods in the `phruta` R package">
<meta property="og:description" content="phruta">
<meta property="og:image" content="https://cromanpa94.github.io/phruta/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phruta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/what_is_phruta.html">What is `phruta`?</a>
    </li>
    <li>
      <a href="../articles/Using_phruta.html">Using the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Exporting_data_phruta.html">To export or not export `phruta` outputs</a>
    </li>
    <li>
      <a href="../articles/Phylogenetics_phruta.html">Phylogenetics with the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Phruta_advanced.html">Advanced methods in the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Future_phruta.html">Future directions of the `phruta` R package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cromanpa94/phruta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Advanced methods in the <code>phruta</code> R
package</h1>
                        <h4 data-toc-skip class="author">Cristian
Román-Palacios</h4>
            
            <h4 data-toc-skip class="date">First created on 2021-07-30.
Updated on 2022-06-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/cromanpa94/phruta/blob/HEAD/vignettes/Phruta_advanced.Rmd" class="external-link"><code>vignettes/Phruta_advanced.Rmd</code></a></small>
      <div class="hidden name"><code>Phruta_advanced.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<ol style="list-style-type: decimal">
<li><a href="#paragraph0">Working with non-animal/plant groups in
<code>phruta</code></a></li>
<li><a href="#paragraph1">Creating taxonomic constraints in
<code>phruta</code></a></li>
<li><a href="#paragraph2">Running PartitionFinder in
<code>phruta</code></a></li>
<li><a href="#paragraph3">Partitioned analyses in RAxML</a></li>
<li><a href="#paragraph4">Identifying rogue taxa</a></li>
</ol>
<div class="section level3">
<h3 id="working-with-non-animal-nor-plant-groups-in-phruta">Working with non-animal nor plant groups in <code>phruta</code>
<a name="paragraph0"></a><a class="anchor" aria-label="anchor" href="#working-with-non-animal-nor-plant-groups-in-phruta"></a>
</h3>
<p>As explained in the brief intro to <code>phruta</code>, the
<code>sq.curate</code> function is primarily designed to curate
taxonomic datasets using <code>gbif</code>. Alto <code>gbif</code> is
extremely fast and efficient, it is largely designed to deal with
animals and plants. If you’re interested in using the <code>gbif</code>
backbone for curating sequence regardless of the kingdom use the
following approach:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/taxonomy.retrieve.html">taxonomy.retrieve</a></span><span class="op">(</span>species_names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis_catus"</span>, <span class="st">"PREDICTED:_Vulpes"</span>,
                  <span class="st">"Phoca_largha"</span>, <span class="st">"PREDICTED:_Phoca"</span> ,
                  <span class="st">"PREDICTED:_Manis"</span> , <span class="st">"Felis_silvestris"</span> , <span class="st">"Felis_nigripes"</span><span class="op">)</span>,
                  database<span class="op">=</span><span class="st">'gbif'</span>, kingdom<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<p>Note that the <code>kingdom</code> argument is set to
<code>NULL</code>. However, as indicated in the first vignette,
<code>gbif</code> is efficient for retrieving accurate taxonomy when we
provide details on the <code>kingdom</code>. Given that all the species
we’re interested in are animals, we could just use:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/taxonomy.retrieve.html">taxonomy.retrieve</a></span><span class="op">(</span>species_names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis_catus"</span>, <span class="st">"PREDICTED:_Vulpes"</span>,
                  <span class="st">"Phoca_largha"</span>, <span class="st">"PREDICTED:_Phoca"</span> ,
                  <span class="st">"PREDICTED:_Manis"</span> , <span class="st">"Felis_silvestris"</span> , <span class="st">"Felis_nigripes"</span><span class="op">)</span>,
                  database<span class="op">=</span><span class="st">'gbif'</span>, kingdom<span class="op">=</span><span class="st">'animals'</span><span class="op">)</span></code></pre></div>
<p>We could also do the same for plants by using <code>plants</code>
instead of <code>animals</code> in the <code>kingdom</code> argument.
Now, what if we were interested in following other databases to retrieve
taxonomic information for the species in our database? The latest
version of <code>phruta</code> allow users to select the desired
database. The databases follow the <code><a href="https://docs.ropensci.org/taxize/reference/classification.html" class="external-link">taxize::classification</a></code>
function. Options are: <code>ncbi</code>, <code>itis</code>,
<code>eol</code>, <code>tropicos</code>, <code>nbn</code>,
<code>worms</code>, <code>natserv</code>, <code>bold</code>,
<code>wiki</code>, and <code>pow</code>. Please select only one. Note
that the <code>gbif</code> option in <code><a href="https://docs.ropensci.org/taxize/reference/classification.html" class="external-link">taxize::classification</a></code>
is replaced by the internal <code>gbif</code> in
<code>phruta</code>.</p>
<p>Now, let’s assume that we were interested in curating our database
using <code>itis</code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/taxonomy.retrieve.html">taxonomy.retrieve</a></span><span class="op">(</span>species_names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis_catus"</span>, <span class="st">"PREDICTED:_Vulpes"</span>,
                  <span class="st">"Phoca_largha"</span>, <span class="st">"PREDICTED:_Phoca"</span> ,
                  <span class="st">"PREDICTED:_Manis"</span> , <span class="st">"Felis_silvestris"</span> , <span class="st">"Felis_nigripes"</span><span class="op">)</span>,
                  database<span class="op">=</span><span class="st">'itis'</span><span class="op">)</span></code></pre></div>
<p>Using alternative databases is sometimes desirable. Please make sure
you review which the best database is for your target group is before
selecting one.</p>
</div>
<div class="section level3">
<h3 id="creating-taxonomic-constraints-in-phruta">Creating taxonomic constraints in <code>phruta</code>
<a name="paragraph1"></a><a class="anchor" aria-label="anchor" href="#creating-taxonomic-constraints-in-phruta"></a>
</h3>
<p>For different reasons, phylogenetic analyses sometimes require of
tree constraints. <code>phruta</code> can automatically generate trees
in accordance to taxonomy and a backbone topology. We divide constraint
trees into two: (1) ingroup+outgroup and (2) particular clades.</p>
<div class="section level4">
<h4 id="ingroup-outgroup">ingroup + outgroup<a class="anchor" aria-label="anchor" href="#ingroup-outgroup"></a>
</h4>
<p>In this constraint type, <code>phruta</code> will create monophyletic
groups for each of the taxonomic groups in the database (for selected
target columns). Finally, it will generate tree with the same topology
provided in the <code>Topology</code> argument. The user will provide
the species names of the outgroup taxa as a vector of string that should
fully match the names in the taxonomy file.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/tree.constraint.html">tree.constraint</a></span><span class="op">(</span>
                taxonomy_folder <span class="op">=</span> <span class="st">"1.CuratedSequences"</span>,
                targetColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kingdom"</span>, <span class="st">"phylum"</span>, <span class="st">"class"</span>, <span class="st">"order"</span>, 
                                  <span class="st">"family"</span>, <span class="st">"genus"</span>, <span class="st">"species_names"</span><span class="op">)</span>,
                Topology <span class="op">=</span> <span class="st">"((ingroup), outgroup);"</span>,
                outgroup <span class="op">=</span> <span class="st">"Manis_pentadactyla"</span>
 <span class="op">)</span></code></pre></div>
<div class="figure">
<img src="IngroupOutgroup.jpg" alt=""><p class="caption">Tree constraints ingroup+outgroup (example shown
based on a much limited number of genes; see the examples in the
package).</p>
</div>
</div>
<div class="section level4">
<h4 id="particular-clades">Particular clades<a class="anchor" aria-label="anchor" href="#particular-clades"></a>
</h4>
<p>In this constraint type, <code>phruta</code> will create a constraint
tree for particular clades. For instance, let’s assume that we only need
to create a tree constraining the monophyly within two genera and their
sister relationships:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/tree.constraint.html">tree.constraint</a></span><span class="op">(</span> taxonomy_folder <span class="op">=</span> <span class="st">"1.CuratedSequences"</span>,
                 targetColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kingdom"</span>, <span class="st">"phylum"</span>, <span class="st">"class"</span>, 
                                   <span class="st">"order"</span>, <span class="st">"family"</span>, <span class="st">"genus"</span>, <span class="st">"species_names"</span><span class="op">)</span>,
                 Topology <span class="op">=</span> <span class="st">"((Felis), (Phoca));"</span>
 <span class="op">)</span></code></pre></div>
<p>Note that the key aspect in here is the <code>Topology</code>
argument. It is a newick tree.</p>
<div class="figure">
<img src="byclade.jpg" alt=""><p class="caption">Tree constraints by clade (example shown based on a
much limited number of genes; see the examples in the package).</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="running-partitionfinder-in-phruta">Running PartitionFinder in <code>phruta</code>
<a name="paragraph2"></a><a class="anchor" aria-label="anchor" href="#running-partitionfinder-in-phruta"></a>
</h3>
<p>With the current version of <code>phruta</code>, users are able to
run PartitionFinder v1 within R. For this, users should provide the name
of the folder where the alignments are stored, a particular pattern in
the file names ( <code>masked</code> in our case), and which models will
be run in PartitionFinder. This function will download PartitionFinder,
generate the input files, and run it all within R. The output files will
be in a new folder within the working directory.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/sq.partitionfinderv1.html">sq.partitionfinderv1</a></span><span class="op">(</span>folderAlignments <span class="op">=</span> <span class="st">"2.Alignments"</span>,
                    FilePatterns <span class="op">=</span> <span class="st">"Masked"</span>,
                    models <span class="op">=</span> <span class="st">"all"</span>
 <span class="op">)</span></code></pre></div>
<p>Unfortunately, the output files are not integrated with the current
<code>phruta</code> pipeline. This will be part of a new release.
However, users can still perform gene-based partitioned analyses within
<code>RAxML</code> or can use PartitionFinder’s output files to inform
their own analyses outside <code>phruta</code>.</p>
</div>
<div class="section level3">
<h3 id="partitioned-analyses-in-raxml">Partitioned analyses in RAxML <a name="paragraph3"></a><a class="anchor" aria-label="anchor" href="#partitioned-analyses-in-raxml"></a>
</h3>
<p>Users can now run partitioned analyses in <code>RAxML</code> within
<code>phruta</code>. This approach is implemented by setting the
<code>partitioned</code> argument in <code>tree.raxml</code> to
<code>TRUE</code>. For now, partitions are based on the genes are being
analyzed. The same model is used to analyze each partition. More details
on partitioned analyses can be customized by passing arguments in
<code><a href="https://rdrr.io/pkg/ips/man/raxml.html" class="external-link">ips::raxml</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/tree.raxml.html">tree.raxml</a></span><span class="op">(</span>folder <span class="op">=</span> <span class="st">"2.Alignments"</span>, FilePatterns <span class="op">=</span> <span class="st">"Masked"</span>,
           raxml_exec <span class="op">=</span> <span class="st">"raxmlHPC"</span>, Bootstrap <span class="op">=</span> <span class="fl">100</span>,
           outgroup <span class="op">=</span> <span class="st">"Manis_pentadactyla"</span>,
           partitioned<span class="op">=</span><span class="cn">T</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="identifying-rogue-taxa">Identifying rogue taxa <a name="paragraph4"></a><a class="anchor" aria-label="anchor" href="#identifying-rogue-taxa"></a>
</h3>
<p><code>phruta</code> can help users run <code>RogueNaRok</code>
implemented in the <code>Rogue</code> R package. Users can then examine
whether rogue taxa should be excluded from the analyses.
<code>tree.roguetaxa</code> uses the bootstrap trees generated using the
<code>tree.raxml</code> function along with the associated best tree to
identify rogue taxa.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/tree.roguetaxa.html">tree.roguetaxa</a></span><span class="op">(</span>folder <span class="op">=</span> <span class="st">"3.Phylogeny"</span><span class="op">)</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
