<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Curate sequences from genbank — sq.curate • phruta</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Curate sequences from genbank — sq.curate"><meta property="og:description" content='After downloading sequences from genbank, this function
curates sequences based on taxonomic
information. Note that this function provides two summary datasets.
First, the accession numbers.
Second, the taxonomic information for each species in the database.
The taxonomy strictly follows
the gbif taxonomic backbone. The resulting files are saved
to "1.CuratedSequences". The
resulting files also have the most recent curated taxonomy
following the gbif (or selected database) taxonomic backbone.'><meta property="og:image" content="https://cromanpa94.github.io/phruta/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phruta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Brief_Intro_phruta.html">Introduction to the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Future_phruta.html">Future directions of the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Phruta_advanced.html">Advanced methods in the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Phylogenetics_phruta.html">Phylogenetics with the `phruta` R package</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/cromanpa94/phruta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Curate sequences from genbank</h1>
    <small class="dont-index">Source: <a href="https://github.com/cromanpa94/phruta/blob/HEAD/R/sq.curate.R" class="external-link"><code>R/sq.curate.R</code></a></small>
    <div class="hidden name"><code>sq.curate.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>After downloading sequences from genbank, this function
curates sequences based on taxonomic
information. Note that this function provides two summary datasets.
First, the accession numbers.
Second, the taxonomic information for each species in the database.
The taxonomy strictly follows
the gbif taxonomic backbone. The resulting files are saved
to <code>"1.CuratedSequences"</code>. The
resulting files also have the most recent curated taxonomy
following the gbif (or selected database) taxonomic backbone.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">sq.curate</span><span class="op">(</span>
  filterTaxonomicCriteria <span class="op">=</span> <span class="cn">NULL</span>,
  mergeGeneFiles <span class="op">=</span> <span class="cn">NULL</span>,
  database <span class="op">=</span> <span class="st">"gbif"</span>,
  kingdom <span class="op">=</span> <span class="cn">NULL</span>,
  folder <span class="op">=</span> <span class="st">"0.Sequences"</span>,
  removeOutliers <span class="op">=</span> <span class="cn">TRUE</span>,
  minSeqs <span class="op">=</span> <span class="fl">5</span>,
  threshold <span class="op">=</span> <span class="fl">0.05</span>,
  ranks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kingdom"</span>, <span class="st">"phylum"</span>, <span class="st">"class"</span>, <span class="st">"order"</span>, <span class="st">"family"</span>, <span class="st">"genus"</span>, <span class="st">"species"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>filterTaxonomicCriteria</dt>
<dd><p>A single string of terms (delimited using "|")
listing all the strings that could be used to
identify the species that should
be in the dataset (character).</p></dd>
<dt>database</dt>
<dd><p>A name of a database with taxonomic information.
Although 'gbif' is faster, it only has information for
animals and plants. Other databases follow
taxize::classification.</p></dd>
<dt>kingdom</dt>
<dd><p>Optional and only used when database='gbif'.
Two possible options: "animals" or "plants."</p></dd>
<dt>folder</dt>
<dd><p>The name of the folder where the original sequences are
located (character).</p></dd>
<dt>removeOutliers</dt>
<dd><p>Whether odseq:odseq should be used to remove outliers</p></dd>
<dt>minSeqs</dt>
<dd><p>minimum number of sequences per locus</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>None</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"><span class="fu">sq.retrieve</span><span class="op">(</span></span>
<span class="r-in">  clades <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis"</span>, <span class="st">"Vulpes"</span>, <span class="st">"Phoca"</span><span class="op">)</span>,</span>
<span class="r-in">  species <span class="op">=</span> <span class="st">"Manis_pentadactyla"</span>,</span>
<span class="r-in">  genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ADORA3"</span>, <span class="st">"CYTB"</span><span class="op">)</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"><span class="fu">sq.curate</span><span class="op">(</span></span>
<span class="r-in">  filterTaxonomicCriteria <span class="op">=</span> <span class="st">"Felis|Vulpes|Phoca|Manis"</span>,</span>
<span class="r-in">  database <span class="op">=</span> <span class="st">"gbif"</span>, kingdom <span class="op">=</span> <span class="st">"animals"</span>,</span>
<span class="r-in">  folder <span class="op">=</span> <span class="st">"0.Sequences"</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer></div>

  


  

  </body></html>

