<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Phylogenetics with the `phruta` R package • phruta</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Phylogenetics with the `phruta` R package">
<meta property="og:description" content="phruta">
<meta property="og:image" content="https://cromanpa94.github.io/phruta/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phruta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/what_is_phruta.html">What is `phruta`?</a>
    </li>
    <li>
      <a href="../articles/Using_phruta.html">Using the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Exporting_data_phruta.html">To export or not export `phruta` outputs</a>
    </li>
    <li>
      <a href="../articles/Phylogenetics_phruta.html">Phylogenetics with the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Phruta_advanced.html">Advanced methods in the `phruta` R package</a>
    </li>
    <li>
      <a href="../articles/Future_phruta.html">Future directions of the `phruta` R package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cromanpa94/phruta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Phylogenetics_phruta_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Phylogenetics with the <code>phruta</code> R package</h1>
                        <h4 data-toc-skip class="author">Cristian Román-Palacios</h4>
            
            <h4 data-toc-skip class="date">First created on 2021-07-30. Updated on 2022-06-27</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/cromanpa94/phruta/blob/HEAD/vignettes/Phylogenetics_phruta.Rmd" class="external-link"><code>vignettes/Phylogenetics_phruta.Rmd</code></a></small>
      <div class="hidden name"><code>Phylogenetics_phruta.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<ol style="list-style-type: decimal">
<li><a href="#paragraph1">Installing RAxML</a></li>
<li><a href="#paragraph2">Installing PATHd-8 and treePL</a></li>
<li><a href="#paragraph3">Phylogenetic inference with <code>phruta</code> and <code>RAxML</code></a></li>
<li><a href="#paragraph4">Tree dating in <code>phruta</code></a></li>
</ol>
<div class="section level3">
<h3 id="installing-raxml">Installing RAxML <a name="paragraph1"></a><a class="anchor" aria-label="anchor" href="#installing-raxml"></a>
</h3>
<p><code>RAxML</code> can be easily installed to the <code>PATH</code> using one of the two lines below in <code>conda</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="ex">conda</span> install -c bioconda/label/cf201901 raxml </span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="ex">conda</span> install -c bioconda raxml</span></code></pre></div>
<p>Open <code>R</code> and make sure that the following line doesn’t throw an error.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/system.html" class="external-link">system</a></span><span class="op">(</span><span class="st">"raxmlHPC"</span><span class="op">)</span></span></code></pre></div>
<p>You may also want to check if <code>RAxML</code> is called from the console using <code>raxmlHPC</code> or <code>raxml</code>. In fact, this string is passed to the <code>exec</code> argument in <code><a href="https://rdrr.io/pkg/ips/man/raxml.html" class="external-link">ips::raxml</a></code> when running <code>tree.raxml</code> (<code>raxml_exec</code> therein).</p>
<p>Finally, note that <code>RStudio</code> sometimes has issues finding <em>stuff</em> in the path while using <code><a href="https://rdrr.io/r/base/system.html" class="external-link">system()</a></code>.</p>
<p>With issues, if you’re running macOS, try running RStudio from the command line. Close RStudio, open the terminal, and type:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1"></a><span class="ex">open</span> /Applications/RStudio.app</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="installing-pathd-8-and-treepl">Installing PATHd-8 and treePL <a name="paragraph2"></a><a class="anchor" aria-label="anchor" href="#installing-pathd-8-and-treepl"></a>
</h3>
<p>There are excellent guides for installing <code>PATHd-8</code> and <code>treePL</code>. Here, I summarize two potentially relevant options.</p>
<p>First, you can use <a href="https://github.com/bomeara/phydocker/blob/master/Dockerfile" class="external-link">Brian O’Meara’s</a> approach for installing <code>PATHd-8</code>. I summarized the code <a href="https://gist.github.com/cromanpa94/a43bc710a17220f71d796d6590ea7fe4" class="external-link">here</a>.</p>
<p>Second, you can use homebrew to install <code>treePL</code>, thanks to Jonathan Chang.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1"></a><span class="ex">brew</span> install brewsci/bio/treepl</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="phylogenetic-inference-with-phruta-and-raxml">Phylogenetic inference with <code>phruta</code> and <code>RAxML</code> <a name="paragraph3"></a><a class="anchor" aria-label="anchor" href="#phylogenetic-inference-with-phruta-and-raxml"></a>
</h3>
<p>Phylogenetic inference is conducted using the <code>tree.raxml</code> function. We need to indicate where the aligned sequences are located (<code>folder</code> argument), the patterns of the files in the same folder (<code>FilePatterns</code> argument; “<code>Masked_</code>” in our case). We’ll run a total of 100 boostrap replicates and set the outgroup to “Manis_pentadactyla”.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tree.raxml.html">tree.raxml</a></span><span class="op">(</span>folder<span class="op">=</span><span class="st">'2.Alignments'</span>, </span>
<span>           FilePatterns<span class="op">=</span> <span class="st">'Masked_'</span>, </span>
<span>           raxml_exec<span class="op">=</span><span class="st">'raxmlHPC'</span>, </span>
<span>           Bootstrap<span class="op">=</span><span class="fl">100</span>, </span>
<span>           outgroup <span class="op">=</span><span class="st">"Manis_pentadactyla"</span><span class="op">)</span></span></code></pre></div>
<p>The trees are saved in <code>3.Phylogeny</code>. Likely, the bipartitions tree, “RAxML_bipartitions.phruta”, is the most relevant. <code>3.Phylogeny</code> also includes additional <code>RAxML</code>-related input and output files.</p>
<p>Finally, let’s perform tree dating in our phylogeny using secondary calibrations extracted from <a href="https://royalsocietypublishing.org/doi/pdf/10.1098/rspb.2016.1334" class="external-link">Scholl and Wiens (2016)</a>. This study curated potentially the most comprenhensive and reliable set of trees to summarize the temporal dimension in evolution across the tree of life. In <code>phruta</code>, the trees from Scholl and Wiens (2016) were renamed to match taxonomic groups.</p>
</div>
<div class="section level3">
<h3 id="tree-dating-in-phruta">Tree dating in <code>phruta</code> <a name="paragraph4"></a><a class="anchor" aria-label="anchor" href="#tree-dating-in-phruta"></a>
</h3>
<p>Tree dating is performed using the <code>tree.dating</code> function in <code>phruta</code>. We have to provide the name of the folder containing the <code>1.Taxonomy.csv</code> file created in <code>sq.curate</code>. We also have to indicate the name of the folder containing the <code>RAxML_bipartitions.phruta</code> file. We will scale our phylogeny using <code>treePL</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tree.dating.html">tree.dating</a></span><span class="op">(</span>taxonomyFolder<span class="op">=</span><span class="st">"1.CuratedSequences"</span>, </span>
<span>            phylogenyFolder<span class="op">=</span><span class="st">"3.Phylogeny"</span>, </span>
<span>            scale<span class="op">=</span><span class="st">'treePL'</span><span class="op">)</span></span></code></pre></div>
<p>Running this line will result in a new folder <code>4.Timetree</code>, including the different time-calibrated phylogenies obained (if any) and associated secondary calibrations used in the analyses. We found only a few overlapping calibration points (family-level constraints):</p>
<p>Here’s the resulting time-calibrated phylogeny. The whole process took ~20 minutes to complete on my computer (16 gb RAM, i5).</p>
<div class="figure">
<img src="tree.png" alt=""><p class="caption">Time-calibrated phylogeny for the target taxa. Analyses performed within the phruta R package.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
